runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java

```diff
@@ -133,36 +138,48 @@ public void updateConfig(ConfigChanged event)
 	@Subscribe
 	public void onChatMessage(ChatMessage event)
 	{
-		if (event.getType() != ChatMessageType.SERVER || !config.showBindNeck())
+		if (event.getType() != ChatMessageType.SERVER)
 		{
 			return;
 		}
 
-		Matcher match = bindNeckString.matcher(event.getMessage());
-		if (match.find())
+		if (config.showBindNeck())
 		{
-			bindNeckOverlay.bindingCharges = Integer.parseInt(match.group(1));
-			return;
-		}
-
-		if (event.getMessage().contains("You bind the temple's power"))
-		{
-			if (event.getMessage().contains("mud")
-				|| event.getMessage().contains("lava")
-				|| event.getMessage().contains("steam")
-				|| event.getMessage().contains("dust")
-				|| event.getMessage().contains("smoke")
-				|| event.getMessage().contains("mist"))
+			Matcher match = bindNeckString.matcher(event.getMessage());
+			if (match.find())
 			{
-				bindNeckOverlay.bindingCharges -= 1;
+				bindNeckOverlay.bindingCharges = Integer.parseInt(match.group(1));
 				return;
 			}
-		}
 
-		if (event.getMessage().contains("Your Binding necklace has disintegrated."))
+			if (event.getMessage().contains("You bind the temple's power"))
+			{
+				if (event.getMessage().contains("mud")
+					|| event.getMessage().contains("lava")
+					|| event.getMessage().contains("steam")
+					|| event.getMessage().contains("dust")
+					|| event.getMessage().contains("smoke")
+					|| event.getMessage().contains("mist"))
+				{
+					bindNeckOverlay.bindingCharges -= 1;
+					return;
+				}
+			}
+
+			if (event.getMessage().contains("Your Binding necklace has disintegrated."))
+			{
+				//set it to 17 because this message is triggered first before the above chat event
+				bindNeckOverlay.bindingCharges = 17;
+			}
+			return;
+		}
+		if (config.degradingNotification())
 		{
-			//set it to 17 because this message is triggered first before the above chat event
-			bindNeckOverlay.bindingCharges = 17;
+			if (event.getMessage().contains("Your pouch has decayed through use."))
```

Comment:
Use a constant for this string. This should probably be `POUCH_DECAYED_MESSAGE`, and the notification string should probably be something like `POUCH_DECAYED_NOTIFICATION_MESSAGE`.

Reply:
Fair enough, wasn't sure as it wasn't done with the binding necklace.